backend/internal/api/service_handlers.go

```diff
@@ -222,14 +223,56 @@ func (h *ServiceHandler) getViewportNodes(c *gin.Context) {
 
 // parseVault triggers a vault parsing operation
 func (h *ServiceHandler) parseVault(c *gin.Context) {
-	// TODO: Implement when VaultService is available
-	c.JSON(http.StatusNotImplemented, gin.H{"error": "Vault parsing not yet implemented"})
+	// Use request context
+	ctx := c.Request.Context()
+
+	// Check if vault service is available
+	if h.vaultService == nil {
+		c.JSON(http.StatusInternalServerError, gin.H{"error": "Vault service not configured"})
+		return
+	}
+
+	// Check if parse is already running
+	status, err := h.vaultService.GetParseStatus(ctx)
+	if err != nil {
+		c.JSON(http.StatusInternalServerError, gin.H{"error": "Failed to get parse status"})
+		return
+	}
+
+	if status.Status == "parsing" {
+		c.JSON(http.StatusConflict, gin.H{"error": "Parse already in progress"})
+		return
+	}
```

Comment:
TODO(CL): Add timeout handling for the parsing operation. For 50k nodes, parsing could take significant time. Consider:
1. Set a context with timeout: `ctx, cancel := context.WithTimeout(ctx, 30*time.Minute)`
2. Make timeout configurable per request or in config
3. Handle context cancellation gracefully
4. Return partial results if timeout occurs